lib/galaxy/managers/pages.py

```diff
@@ -294,6 +296,40 @@ def create_page(self, trans, payload: CreatePagePayload):
         session.commit()
         return page
 
+    def update_page(self, trans, id: PageIdPathParam, payload: UpdatePagePayload):
+        user = trans.get_user()
+
+        # Load page from database
+        page = trans.sa_session.get(model.Page, id)
+        if not page:
+            raise exceptions.ObjectNotFound("Page not found")
+        page = base.security_check(trans, page, check_ownership=False, check_accessible=True)
+
+        # Validate payload
+        if not payload.title:
```

Comment:
Set a minimum length in the pydantic model, it'll then be documted properly and you don't need manual validation. Same for slug, and you can already set a regex for the slug in the model, it'll also become part of the API documentation. Finally the annotation does not need to be html sanitized if you are not using `v-html` (I haven't checked that you are not!).